src/views/open-api-docs-view/components/open-api-sidebar-contents/utils/use-nav-items-with-active.ts

```diff
@@ -0,0 +1,105 @@
+// Third-party
+import { useRouter } from 'next/router'
+import { useMemo } from 'react'
+// Lib
+import { useActiveSection } from 'lib/hash-links/use-active-section'
+import newUrl from 'lib/new-url'
+// Types
+import type { OpenApiNavItem } from 'views/open-api-docs-view/types'
+
+/**
+ * Note next/router `asPath` returns *with* the `#hash`.
+ *
+ * Here, we remove the hash, so that we can compare to the `fullPath`
+ * of each nav item, which are not expected to have hashes,
+ * to the URL path that we're on.
+ *
+ * Note that domain does not matter, as we're just grabbing the full path.
+ */
+function usePathname() {
+	const { asPath } = useRouter()
+	return useMemo(() => newUrl(asPath).pathname, [asPath])
```

Comment:
❓ I'm surprised this wouldn't error without the base path as the second arg to the url constructor? Isn't `asPath` just a relative path?